eng/pipelines/runtime.yml

```diff
@@ -51,6 +51,12 @@ variables:
   - template: /eng/pipelines/common/variables.yml
 
 jobs:
+
+#
+# Source Index Build
+#
+- template: ./source-index.yml
```

Comment:
Should we use the path filters for this as well? I presume that we only need to run this if C# code is changed and only in some code paths?

Reply:
If its not difficult that would be fine. What do I put where to do that?

Reply:
You can add a subset here: https://github.com/dotnet/runtime/blob/master/eng/pipelines/common/evaluate-default-paths.yml#L66 and then you can condition the job based on that subset, something like this:https://github.com/dotnet/runtime/blob/c9d1fd60d6df9c887a1ec5f645ce57dbc5cd47e1/eng/pipelines/runtime.yml#L82-L86

Note that this only applies for CI, official builds doesn't filter per path.

Reply:
I can't do this without more changes in arcade. I can't change this job to depend on something else.

Reply:
Can we add the right parameters to arcade, like `dependsOn` and `condition`? I'd like to avoid running this job in cases where it is not needed as we already use many resources.

We can merge as is, and follow up, but please open an issue.

Reply:
Yea, I can remove the part running this in PRs, then add the parameters to arcade, and put the PR stuff back.

Reply:
Sounds like a good plan.